File: backend/src/main/java/com/optimiza/clickbarber/controller/BarbeariaController.java
package com.optimiza.clickbarber.controller;

import com.optimiza.clickbarber.model.Resposta;
import com.optimiza.clickbarber.model.RespostaUtils;
import com.optimiza.clickbarber.model.dto.barbearia.BarbeariaDto;
import com.optimiza.clickbarber.service.BarbeariaService;
import com.optimiza.clickbarber.utils.Constants;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/barbearias")
public class BarbeariaController {

    private final BarbeariaService barbeariaService;

    @Autowired
    public BarbeariaController(BarbeariaService barbeariaService) {
        this.barbeariaService = barbeariaService;
    }

    @GetMapping
    public Resposta<List<BarbeariaDto>> buscarPorNome(@RequestParam(name = "nome") String nome) {
        var barbearias = barbeariaService.buscarPorNome(nome);
        var mensagem = Constants.Success.BARBEARIAS_ENCONTRADAS_PELO_NOME + nome;
        return RespostaUtils.ok(mensagem, barbearias);
    }

}

File: backend/src/main/java/com/optimiza/clickbarber/controller/BarbeariaController.kt
package com.optimiza.clickbarber.controller

import com.optimiza.clickbarber.model.Resposta
import com.optimiza.clickbarber.model.RespostaUtils
import com.optimiza.clickbarber.model.dto.barbearia.BarbeariaDto
import com.optimiza.clickbarber.service.BarbeariaService
import com.optimiza.clickbarber.utils.Constants
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/barbearias")
class BarbeariaController @Autowired constructor(
    private val barbeariaService: BarbeariaService
) {

    @GetMapping
    fun buscarTodos(@RequestParam(name = "limit") limit: Int = 0): Resposta<List<BarbeariaDto>> {
        val pageable = if (limit == null || limit == 0) {
            Pageable.unpaged()
        } else {
            PageRequest.of(0, limit)
        }

        val barbeariasEncontradas = barbeariaService.buscarTodos(pageable)
        val mensagem = "${barbeariasEncontradas.size} ${Constants.Success.BARBEARIAS_ENCONTRADAS}"

        return RespostaUtils.ok(mensagem, barbeariasEncontradas)
    }

    @GetMapping("/nome")
    fun buscarPorNome(@RequestParam(name = "nome") nome: String): Resposta<List<BarbeariaDto>> {
        val barbearias = barbeariaService.buscarPorNome(nome)
        val mensagem = "${Constants.Success.BARBEARIAS_ENCONTRADAS_PELO_NOME} $nome"
        return RespostaUtils.ok(mensagem, barbearias)
    }

}